src/PaymentApi.php

```diff
@@ -160,140 +113,86 @@ public function debitTransaction( $transactionId, Transaction $transaction )
 
-        if(Validator::intVal()->max(100, true)->notEmpty()->validate($limit)){
-            $uri += '&limit='.$limit;
+        if (Validator::intVal()->max(100, true)->notEmpty()->validate($limit)) {
+            $uri .= '&limit=' . $limit;
         }
-        if(Validator::date("Y-m-d'T'HH:mm:ss'Z'")->notEmpty()->validate($startDate)){
-            $uri += '&start-date='.urlencode($startDate);
+        if (Validator::date("Y-m-d'T'HH:mm:ss'Z'")->notEmpty()->validate($startDate)) {
+            $uri .= '&start-date=' . urlencode($startDate);
         }
-        if(Validator::date("Y-m-d'T'HH:mm:ss'Z'")->notEmpty()->validate($endDate)){
-            $uri += '&end-date='.urlencode($limit);
+        if (Validator::date("Y-m-d'T'HH:mm:ss'Z'")->notEmpty()->validate($endDate)) {
+            $uri .= '&end-date=' . urlencode($limit);
         }
 
-        ksort($headers);
-
-        $signature = $this->createSecureSign(self::$METHOD_GET, $uri, $headers);
-
-        $headers[self::$SIGNATURE] = $signature;
-        $headers[self::$CT_HEADER] = self::$CT_HEADER_INFO;
-
-        $response = Request::get($this->serviceUrl . $uri)
-            ->addHeaders($headers)
-            ->send();
-
-        return $response;
+        return $this->makeRequest(self::$METHOD_GET, $uri);
     }
 
     /**
      * @param $tokenizeId
      * @return \Httpful\Response
      */
-    public function tokenize( $tokenizeId )
+    public function tokenize($tokenizeId)
     {
-        $headers = $this->createHeaderNameValuePairs();
         $uri = '/tokenization/' . $tokenizeId;
-
-        ksort($headers);
-
-        $signature = $this->createSecureSign(self::$METHOD_GET, $uri, $headers);
-
-        $headers[self::$SIGNATURE] = $signature;
-        $headers[self::$CT_HEADER] = self::$CT_HEADER_INFO;
-
-        $response = Request::get($this->serviceUrl . $uri)
-            ->addHeaders($headers)
-            ->send();
-
-        return $response;
+        return $this->makeRequest(self::$METHOD_GET, $uri);
     }
 
     /**
      * @param string $date date in format yyyyMMdd
      * @return \Httpful\Response
      * @throws \Httpful\Exception\ConnectionErrorException
      */
-    public function getReport( $date )
+    public function getReport($date)
     {
-        $headers = $this->createHeaderNameValuePairs();
         $uri = '/report/batch/' . $date;
+        return $this->makeRequest(self::$METHOD_GET, $uri);
+    }
 
-        ksort($headers);
-
-        $signature = $this->createSecureSign(self::$METHOD_GET, $uri, $headers);
-
-        $headers[self::$SIGNATURE] = $signature;
-        $headers[self::$CT_HEADER] = self::$CT_HEADER_INFO;
-
-        $response = Request::get($this->serviceUrl . $uri)
-            ->addHeaders($headers)
-            ->send();
 
-        return $response;
+    /**
+     * @param string $date in format yyyyMMdd. The date to fetch the reconciliation report for. Must be today - 1 day or earlier.
+     * @param bool $useDateProcessed
```

Comment:
In comment it could say, use something like "Use the acquirer processed date instead of report received date. Might cause changes to the past"